meteor/sofie-server/data-access/repositories/mongo/base-mongo-repository.ts

```diff
@@ -0,0 +1,28 @@
+import { MongoDatabase } from './mongo-database'
+import { MongoEntityConverter } from './mongo-entity-converter'
+import { Collection } from 'mongodb'
+import { DatabaseNotConnectedException } from '../../../model/exceptions/database-not-connected-exception'
+
+export abstract class BaseMongoRepository {
+	protected mongoDatabase: MongoDatabase
+	protected mongoEntityConverter: MongoEntityConverter
+
+	protected constructor(mongoConnection: MongoDatabase, mongoEntityConverter: MongoEntityConverter) {
+		this.mongoDatabase = mongoConnection
+		this.mongoEntityConverter = mongoEntityConverter
+	}
+
+	protected abstract getCollectionName(): string
+
+	protected getCollection(): Collection {
+		return this.mongoDatabase.getCollection(this.getCollectionName())
+	}
+
+	protected assertDatabaseConnection(queryName: string): void {
+		if (!this.getCollection()) {
+			throw new DatabaseNotConnectedException(
+				`Unable to perform query: ${queryName} - not connected to database.collection: ${this.mongoDatabase.getDatabaseName()}.${this.getCollectionName()}`
+			)
+		}
+	}
```

Comment:
The reason is twofold.

First is to have some sort of (in this case) very basic validation if we still had a connection to the database (this might not be the best way to do it.)

The second is that I would like us to start utilizing exceptions. The reasons for this are too many to start writing them here, but in essence we would be able to get stacktraces and descriptive error messages which makes debugging way easier and we would be able to stop parsing undefined and null values around.

Reply:
But right now the return type of `getCollection` is `Collection`, so will the exception ever be thrown here?

Reply:
I have seen it thrown. 

If the variable has never been instantiated then it's still an undefined value.
In the code below an undefined exception is thrown because `getConnection` returns `undefined` even though its return type is `Connection` since it has never been instantiated:
```
class SomeClass {
  private connection Connection

  public getConnection(): Connection {
    return this.connection
  }
}

new SomeClass().getConnection().find({})
```

This is why there are a lot of linting rules forcing non-optional instance variable to be set immediatly or in the constructor.

Reply:
But I agree, there definately is a better way to check if we still have a connection